src/open_inwoner/openklant/services.py

```
@@ -136,9 +136,17 @@ def get_fetch_parameters(
 
         return None
 
+    @property
+    def supports_anonymous_questions(self):
+        if isinstance(self.config, ESuiteKlantConfig):
+            return True
+        if isinstance(self.config, OpenKlant2Config):
+            return False
```

Here the "match and raise" pattern would also be appropriate I think: don't want to silently fail (in that case `None` would be return, and if we ever used `is True` we'd be in trouble).

---

I agree about `raise`, but I found it difficult to work with `match` in this situation. If you do `match type(foo)...case(bar):...do stuff` you'll get `SyntaxError: name capture 'bar' makes remaining patterns unreachable`. I stuck with the `if...else` therefore. Though perhaps I'm missing an obvious way to do this with `match`.

---

The syntax for instance checks would be:

```python
match self.config:
    case OpenKlant2Config():  # equivalent to isinstance(self.config, OpenKlant2Config)
         ...
    case _:
        raise RuntimeError("Unhandled case")
```